### DISCUSSION TITLE
What's the alternative of mlx_clear_window ?

### DISCUSSION THREAD

**Discussion by s0uk1** [codam-coding-college/MLX42#93] [Q&A]
Im working on a wolfenstein3D clone project and while looping i thought i should clear the window and redraw my frame when i move the player or rotate the field of view , is there a way to do it using this version of MLX ?

**Comment by W2Wizard** [comment]
MLX42 has a fundementally different concept of rendering compared to the other miniLibX, it's one if not the only major differences in terms of the API.

In this case the rendering is partially taken care for you. Instead of you doing the updating every time you change the buffer, it is updated automatically. This means you just push the image to the window once, change the buffer at any point in time, the image is updated.

`Why?` you may ask ? Because it is fundementally easier and more convenient to work with. You just change the buffer and it just works. This however means you need to manage your images properly, the idea of a clear window is no longer necessary since you can just implement it yourself by using `memset()` on the buffer with `0`.

Also be aware that you do not need to delete the image and create a new one every time you move, thats just cursed, just update the buffer and thats it.

**Reply by W2Wizard** [reply]
Here is a little demonstration of what I mean, you can just put the image onto the window and update whenever you need to.

```c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <unistd.h>
#include <MLX42/MLX42.h>

int get_rgba(int r, int g, int b, int a)
{
    return (r << 24 | g << 16 | b << 8 | a);
}

void ft_randomize(void* img)
{
	mlx_image_t* image = (mlx_image_t*)img;

	for (int i = 0; i < image->width; ++i)
	{
		for (int y = 0; y < image->height; ++y)
		{
			uint32_t color = get_rgba(rand() % 0xFF, rand() % 0xFF, rand() % 0xFF, 0xFF);
			mlx_put_pixel(image, i, y, color);
		}
	}
}

int main(int argc, char const *argv[])
{
	mlx_t* mlx = mlx_init(420, 420, "42", true);
	if (!mlx)
	{
		puts(mlx_strerror(mlx_errno));
		return (EXIT_FAILURE);
	}
	
	mlx_image_t* image = mlx_new_image(mlx, 420, 420);
	mlx_image_to_window(mlx, image, 0, 0);

	mlx_loop_hook(mlx, ft_randomize, image);
	mlx_loop(mlx);
	mlx_terminate(mlx);
	return (EXIT_SUCCESS);
}
```
